ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/transaction/jts/jca/XATerminatorImple.java

```diff
@@ -452,7 +453,12 @@ public boolean beforeCompletion (Xid xid) throws javax.transaction.SystemExcepti
 
     public Transaction getTransaction(Xid xid) throws XAException {
         // first see if the xid is a root coordinator
-        return com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple.getTransaction(new XidImple(xid).getTransactionUid());
+        Transaction transaction = TransactionImple.getTransaction(new XidImple(xid).getTransactionUid());
```

Comment:
OK, I see why it works. That first call will always be null but it will drop into  SubordinationManager.getTransactionImporter().getImportedTransaction(xid); which should pick it out. It would be clearer to implement (for the JTS version) just as the  SubordinationManager.getTransactionImporter().getImportedTransaction(xid); call